tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+```sh
+(girder)$ girder-worker -l info
+
+ -------------- celery@host v4.0.2
+---- **** -----
+--- * ***  * -- Linux-4.8.6-1-ARCH-x86_64-with-glibc2.2.5
+-- * - **** ---
+- ** ---------- [config]
+- ** ---------- .> app:         girder_worker:0x7f69bfff1050
+- ** ---------- .> transport:   amqp://guest:**@localhost:5672//
+- ** ---------- .> results:     amqp://
+- *** --- * --- .> concurrency: 32 (prefork)
+-- ******* ----
+--- ***** ----- [queues]
+ -------------- .> celery           exchange=celery(direct) key=celery
+
+
+[tasks]
+  . girder_worker.convert
+  . girder_worker.run
+  . girder_worker.validators
+  . gwexample.analyses.tasks.fibonacci
+
+[2016-11-08 12:22:56,163: INFO/MainProcess] Connected to amqp://guest:**@127.0.0.1:5672//
+[2016-11-08 12:22:56,184: INFO/MainProcess] mingle: searching for neighbors
+[2016-11-08 12:22:57,198: INFO/MainProcess] mingle: all alone
+[2016-11-08 12:22:57,218: WARNING/MainProcess] celery@host ready.
+```
+
+
+## What if my task has complex install dependencies? 
+
+One of the few disadvantages of the Celery method is requiring that the package that contains your task be installed on both the consumer (where the task is run) and the producer (where the task is enqueued). Normally this isn't a big problem, especially if your tasks are only using functions from Python's standard library. 
+
+In some cases however, tasks have complex dependencies that are difficult to install and it may be burdensome to install them in the producer (Girder) environment. Luckily this issue can be easily ameliorated by guarding imports in your task packages:
+
+```python
+from girder_worker.app import app
+
+try:
+	from complex_dependency import complex_function
+except ImportError:
+	pass
+
+@app.task
+def some_task(*args, **kwargs):
+	return complex_function(*args, **kwargs
+
+```
+
+Producer-side code will still generate the task objects, allowing them to produce task messages,  while consumer-side code (where the complex_dependency package must be installed) will be able to execute the task body.
+
+You may additionally wish to configure setuptools using [extras_require](http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies) to manage the different between producer and consumer dependencies:
+
+```python
+setup(
+    name="my_custom_package",
+    ...
+	install_requires=[
+		"girder_worker",
+		....
+	],
+    extras_require={
+        'consumer':  ["complex_dependency", ...],
+    }
+	...
+)
+```
+
+This means the producer side code can be installed with pip as ```my_custom_package``` and the consumer side code (that includes the complex dependencies) can be installed as ```my_custom_package[consumer]```
+
+## Communicating with Girder from inside Tasks
+
+Tasks decorated with the Girder Worker application object have two main channels of interacting with the Girder instance that produced the task. 
```

Comment:
when you say `Girder instance`, you mean something like `set of Girder instances tied to the same database` or maybe `Girder host`.  Pedantically, the way you saying isn't correct in a multi-app server Girder environment.